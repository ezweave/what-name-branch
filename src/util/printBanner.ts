import { Logger } from '@ezweave/types';
import chalk from 'chalk';
import figlet from 'figlet';
import { flow } from 'lodash';

interface PrintBanner {
  (log?: Logger): void
}

export const printBanner: PrintBanner = (log = console.log) => flow(
  chalk.bgBlue,
  text => figlet.textSync(text, { horizontalLayout: 'full' }),
  log,
)('what-name-branch?');
