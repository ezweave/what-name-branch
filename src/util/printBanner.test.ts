import { printBanner } from './printBanner';

describe(printBanner, () => {
  it('prints the banner', () => {
    const logger = jest.fn();
    printBanner(logger);
    const call = logger.mock.calls[0][0];
    expect(call).toBeTruthy(); // the snapshot ends up different on the GitHub action runner, so...
  });
});
