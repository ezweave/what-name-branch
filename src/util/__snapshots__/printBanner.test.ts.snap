// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printBanner prints the banner 1`] = `
"[44m           _           _                                  ___ [49m
[44m __      _| |__   __ _| |_      _ __   __ _ _ __ ___   __|__ \\\\[49m
[44m \\\\ \\\\ /\\\\ / / '_ \\\\ / _\` | __|____| '_ \\\\ / _\` | '_ \` _ \\\\ / _ \\\\/ /[49m
[44m  \\\\ V  V /| | | | (_| | ||_____| | | | (_| | | | | | |  __/_| [49m
[44m   \\\\_/\\\\_/ |_| |_|\\\\__,_|\\\\__|    |_| |_|\\\\__,_|_| |_| |_|\\\\___(_) [49m
[44m                                                              [49m"
`;
